File: homework/45278/js2html/js2html/Models/Token.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace js2html.Models
{
    public class Token
    {
        public TokenType TokenType { get; set; }
        public int Line { get; set; }
        public string Lexeme { get; set; }
        public double NumericValue { get; set; }

        public Token(TokenType tokenType, int line, string lexeme = "", double numericValue = 0 )
        {
            TokenType = tokenType;
            Line = line;
            Lexeme = lexeme;
            NumericValue = numericValue;
        }
    }
}

File: homework/81333/golang-highlighter/src/Data/Token.hs
module Data.Token where

import qualified Data.Text       as T
import           Data.TokenTypes (TokenType)

type Lexeme = T.Text

type Line = Int

type Column = Int

data Token = Token
  { tokenType :: TokenType
  , lexeme    :: Lexeme
  , location  :: (Line, Column)
  } deriving (Show)
